### PR TITLE
Adding JCSMP client channel reconnect properties.

### PR DESCRIPTION
**Pull request by dishantlangayan**
Added following 4 properties that control auto reconnect behaviour:
- ConnectRetries
- ReconnectRetries
- ConnectRetriesPerHost
- ReconnectRetriesWaitInMillis

### CODE REVIEW

File: solace-java-spring-boot-autoconfigure/src/main/java/com/solace/labs/spring/boot/autoconfigure/SolaceJavaProperties.java [hunk 1]
```diff
@@ -66,6 +66,31 @@
...
+     */
+    private int connectRetriesPerHost = 0; 
+    
+    /**
+     * How much time in (MS) to wait between each attempt to connect or reconnect to a host. 
+     * If a connect or reconnect attempt to host is not successful, the API waits for the 
+     * amount of time set for reconnectRetryWaitInMillis, and then makes another connect or 
+     * reconnect attempt.
+     */
+    private int reconnectRetryWaitInMillis = 3000;
```
Comment (mdspielman) [inline solace-java-spring-boot-autoconfigure/src/main/java/com/solace/labs/spring/boot/autoconfigure/SolaceJavaProperties.java]: Can you set these to the values recommended by Solace documentation: http://docs.solace.com/Solace-Messaging-APIs/Configuring-Connection-T.htm?Highlight=connectRetries

It's probably good to include that link as well in the description of why you're choosing these values.

**Review by mdspielman — CHANGES_REQUESTED** [review]
I'd like to set the retries to the suggested values from Solace documentation.

File: solace-java-spring-boot-autoconfigure/src/main/java/com/solace/labs/spring/boot/autoconfigure/SolaceJavaProperties.java (hunk 1 above)
Comment (dishantlangayan) [reply solace-java-spring-boot-autoconfigure/src/main/java/com/solace/labs/spring/boot/autoconfigure/SolaceJavaProperties.java]: I think that is a good idea. Will update to the recommended value and also update the unit test.

Comment (mdspielman) [reply solace-java-spring-boot-autoconfigure/src/main/java/com/solace/labs/spring/boot/autoconfigure/SolaceJavaProperties.java]: Excellent

**Comment by dishantlangayan** [comment]
Suggestions have been completed in 2 commits. In 2nd commit I reworded the description for the reconnect properties. Also, there was a copy and paste error on the license txt in the SolaceJavaProperties.java file from the original author, so I fixed that as well.

**Comment by mdspielman** [comment]
Thanks Dishant.

**Pull request merged**